Update Viadotto Acerno to IFC4X3_ADD2

michelangelo-acca opened the pull request:
Update IFC model of Viadotto Acerno to IFC4X3_ADD2 and make it compliant with bSI Validation Service checks.

michelangelo-acca committed:
Update Viadotto Acerno to IFC4X3_ADD2

Update IFC model of Viadotto Acerno to IFC4X3_ADD2 and make it compliant with bSI Validation Service checks.

ptrsmo reviewed (changes requested):
The ifc file should be updated to include the pylons and piers.

michelangelo-acca commented on IFC-files/ACCA/Bridge-for-MINnD/Viadotto Acerno.zip:
Hi @piotr-smolira, I am not having any problem in visualizing the file correctly.

You can have a look at it [here ](https://service.usbim.com/link/gKK31d3dD12ZKxk3Umb5hAXX) or you can upload it on your own using our (free) cloud service, [usBIM](https://cloud.usbim.com).

Let me know if you spot any mistake!

piotr-smolira replied:
Hi @michelangelo-acca 

It appears that indeed the model is complete. It might be shortcoming of viewers.
This is how it looks in BIM Zoom Collab: 
![image](https://github.com/buildingSMART/IFC4.x-IF/assets/55686155/84b885c6-0817-4c4f-802a-6a6cc041fb8a)

This is Open IFC viewer:
![image](https://github.com/buildingSMART/IFC4.x-IF/assets/55686155/364dc82c-258d-423e-9d59-41defb550ad6)

This is how it looks when I uploaded file to cloud.usBIM and view in a viewer:
![image](https://github.com/buildingSMART/IFC4.x-IF/assets/55686155/094ebba8-6972-4bc6-92fd-779749d26846)


This is how it looks in desktop usBIM viewer:
![image](https://github.com/buildingSMART/IFC4.x-IF/assets/55686155/068dabff-2bc7-4aff-91d9-c048abaf5ba6)



I wonder if it something with ACCA setting or some other reason. Because from the link it just works perfectly fine. Only viewer that displays everything and quite fast.



In solibri it doesn't open.

aothms replied:
IfcOpenShell (latest of the v0.8 branch) renders the model like this:

![afbeelding](https://github.com/buildingSMART/IFC4.x-IF/assets/1096535/829f5512-adc2-4a1d-bd84-0475f33d0ee3)

I'm not saying IfcOpenShell is doing the correct thing, things are under heavy development still, but it seems the model is complete.

peterrdf replied:
I am getting a similar image in IFC Viewer from RDF, including the alignments in the other direction. Also there seem to be three IfcSurfaceFeatures that are flying around in the file.
![image](https://github.com/buildingSMART/IFC4.x-IF/assets/6512688/20862a29-40a0-4c02-be66-a332775cad1b)

piotr-smolira replied:
@peterrdf It looks fine in rdf viewer.
@aothms could you pass me bbim with ifcopenshell 0.8? I would love to try it. I'm currently sitting on latest release of 0.7.

aothms replied:
> could you pass me BBIM with IfcOpenShell 0.8

I doesn't exist yet, but I think we're close.

If you want to try ahead of the pack where it fails you can always replace the ifcopenshell folder in your bbim installation with https://github.com/IfcOpenBot/IfcOpenShell/commit/32308540fc57f4e44c6fbac8d4e9bd27b4fd04f7#commitcomment-140952402 and see where it fails. I'll be around to advise.

michelangelo-acca replied:
@peterrdf @aothms Glad to see the model is opening correctly in different viewers!

Regarding the IfcSurfaceFeatures, they do not have ContainedInStructure relationship, as it is now allowed by the schema, but they do have the AdheresToElement one.

aothms replied:
@piotr-smolira shall we move that discussion to somewhere in the ifcopenshell repo as it's not relevant to the people here and we can benefit from @Andrej730 and @Moult's expertise.

michelangelo-acca committed:
Update Viadotto Acerno

Update Viadotto Acerno IFC model, now 100% compliant with bSI Validator latest version (0.6.2).

michelangelo-acca commented:
Just pushed an updated IFC model, now 100% compliant with the latest bSI Validator (0.6.2)!

michelangelo-acca commented:
P.S.: there is this rule here that probably needs to be updated: 

https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcCorrectLocalPlacement.htm

It returns "?" when, like in the example, there are objects with local placements that are positioned relative to objects with linear placements (like in the case of the assemblies of the piles).

pjanck commented:
> there is this rule here that probably needs to be updated:

A long known issue: https://github.com/buildingSMART/IFC4.3.x-development/issues/615

jeangual commented:
Hello everybody ,

I loaded the model in our application but I modified a field in the definition of clothoids which was incomplete from my point of view (see the example of ifcGradientCurve from BSI). I attach screenshots and the modified ifc file.
[ViadottoAcerno.zip](https://github.com/user-attachments/files/15805837/ViadottoAcerno.zip)
[view.pdf](https://github.com/user-attachments/files/15805897/view.pdf)
